x-pack/plugins/security_solution/public/management/components/console/hooks/use_storage_key.ts

```diff
@@ -0,0 +1,17 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useMemo } from 'react';
+import { useConsoleStore } from '../components/console_state/console_state';
+
+export const useStorageKey = (suffix: string): string | undefined => {
+  const prefix = useConsoleStore().state.storagePrefix;
+
+  return useMemo(() => {
+    return prefix ? `${prefix}.${suffix}` : undefined;
```

Comment:
Should we return just the `suffix` if the `prefix` doesn't exists?

Reply:
No. The intent here is to not save content to local storage if the consumer of the component (`<Console/>`) had defined a prefix for local storage key. If they don't then we don't save it out